pkg/plugin/proto/converter_test.go

```diff
@@ -0,0 +1,251 @@
+				Kind:              "test-kind",
+				EmbodiedEmissions: v1.ResourceEmissions{Value: 200, Unit: "test-unit"},
+				Labels:            map[string]string{"label1": "value1", "label2": "value2"},
+				Metrics: map[string]v1.Metric{
+					"metric1": {
+						Name:       "metric1",
+						Usage:      100,
+						UnitAmount: 10.5,
+						Unit:       "test-unit",
+						Emissions:  v1.ResourceEmissions{Value: 50, Unit: "test-unit"},
+						Labels:     map[string]string{"label1": "value1"},
+						UpdatedAt:  time.Now(),
+					},
+				},
+			},
+			expected: &InstanceRequest{
+				Provider:          "test",
+				Service:           "test-service",
+				Name:              "test-instance",
+				Region:            "test-region",
+				Zone:              "test-zone",
+				Kind:              "test-kind",
+				EmbodiedEmissions: &ResourceEmissions{Value: 200, Unit: "test-unit"},
+				Labels:            map[string]string{"label1": "value1", "label2": "value2"},
+				Metrics: map[string]*Metric{
+					"metric1": {
+						Name:       "metric1",
+						Usage:      100,
+						UnitAmount: 10.5,
+						Unit:       "test-unit",
+						Emissions:  &ResourceEmissions{Value: 50, Unit: "test-unit"},
+						Labels:     map[string]string{"label1": "value1"},
+						UpdatedAt:  time.Now().Unix(),
+					},
+				},
+			},
+		},
+		// Add more test cases as needed
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			result, err := ConvertToPB(test.src)
+			if err != nil {
+				t.Fatalf("ConvertToPB failed with error: %v", err)
+			}
+
+			// Compare result with expected
+			if !compareInstanceRequests(result, test.expected) {
+				t.Errorf("ConvertToPB result does not match expected")
+			}
+		})
+	}
+}
+
+func compareInstanceRequests(a, b *InstanceRequest) bool {
+	if (a == nil && b != nil) || (a != nil && b == nil) {
+		return false
+	}
+	if a == nil && b == nil {
+		return true
+	}
+
+	// Compare exported fields
+	if a.Provider != b.Provider || a.Service != b.Service || a.Name != b.Name ||
+		a.Region != b.Region || a.Zone != b.Zone || a.Kind != b.Kind {
+		return false
+	}
+
+	// Compare EmbodiedEmissions
+	if !compareResourceEmissions(a.EmbodiedEmissions, b.EmbodiedEmissions) {
+		return false
+	}
+
+	// Compare Labels
+	if !compareStringMaps(a.Labels, b.Labels) {
+		return false
+	}
+
+	// Compare Metrics
+	if !compareMetricsMap(a.Metrics, b.Metrics) {
+		return false
+	}
+
+	return true
+}
+
+func compareResourceEmissions(a, b *ResourceEmissions) bool {
+	if a == nil && b != nil || a != nil && b == nil {
+		return false
+	}
+	if a == nil && b == nil {
+		return true
+	}
+
+	return a.Value == b.Value && a.Unit == b.Unit
+}
+
+func compareStringMaps(a, b map[string]string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for key, valueA := range a {
+		valueB, ok := b[key]
+		if !ok || valueA != valueB {
+			return false
+		}
+	}
+
+	return true
+}
+
+func compareMetricsMap(a, b map[string]*Metric) bool {
```

Comment:
Why not use `reflect.DeepEqual` for comparing?

Reply:
There are some fileds in the struct that are unique (locks, mutex, unexported etc) and because of this using deep equal does not work as the structs will never truly be equal